### ISSUE TITLE
Package: Separate version and metadata

### ISSUE THREAD

**Issue by mathias-lang-sociomantic** [sociomantic-tsunami/makd#14]
Currently a user wishing to add metadata to a package has little means to do it.

My use case:
I have an app (`myapp`) that can compiler both in D1 and D2. I'd like this information to reflect in the package version, without affecting the precedence on version.

According to semver.org this kind of information should go into the metadata, which is provided "[...] by appending a plus sign and a series of dot separated identifiers immediately following the patch or pre-release version."
So if Makd generate the version "1.42.1+dirty.20160804", I'd be able to turn it into "1.42.1+D2.dirty.20160804"

I tried using the prefix, but that result in `dpkg` seeing 2 different packages. `myapp` and `myapp-D2` and they obviously conflict because they read /write from the same location.

**Comment by leandro-lucarella-sociomantic** [comment]
I think all in all, using the suffix might be the best solution for this. When supporting suffix properly, you should name files differently too (adding the suffix to files too). If you have configuration files that should be shared (and they shouldn't have the suffix) you can create a `myapp-common` package with them.

Another problem is semver scheme is incompatible with Debian packages version scheme, so translating versions is not trivial. But if all you need is to add something after the `+` and before the `dirty` marker, I guess it could be added.

Also, have you tried redefining `PKGVERSION`? It's declared using `:=` but the `Build.mak` is included  after it, so it should be able to override it.

**Comment by mathias-lang-sociomantic** [comment]
> I think all in all, using the suffix might be the best solution for this. When supporting suffix properly, you should name files differently too (adding the suffix to files too). If you have configuration files that should be shared (and they shouldn't have the suffix) you can create a `myapp-common` package with them.

In the current setup, I don't ever want both package installed at the same time. I think it has value to note that this specific application is compiled in D2, but it shouldn't be its own package.

> Also, have you tried redefining `PKGVERSION` ?

Not yet. I settled on getting rid of the D2 mention for the time being, and seeing if we can find a better solution. But that would require me to either append at the end or parse `PKGVERSION` for `~` right ?

**Comment by leandro-lucarella-sociomantic** [comment]
Yes, it will involve either parsing or building the version yourself. You could copy&paste the code in Makd for the latter. I know, not ideal, but it seems like a fairly obscure use case to add. I consider Makd provides an automatic version number and if you want to go beyond that, you should just provide your custom version number yourself.

**Comment by leandro-lucarella-sociomantic** [comment]
But if you try it, please report if it works or not :)
